libbeat/common/mapval/is_defs.go

```diff
@@ -30,6 +31,29 @@ var KeyPresent = IsDef{name: "check key present"}
 // KeyMissing checks that the given key is not present defined.
 var KeyMissing = IsDef{name: "check key not present", checkKeyMissing: true}
 
+// IsStringContaining validates that the the actual value contains the specified substring.
+func IsStringContaining(needle string) IsDef {
```

Comment:
how about `IsStringMatching`?

Using the FP approach we can also provide predicates:

```
func IsStringWith(pred func(str string) bool) IsDef {
```

Reply:
I think this is a great idea, but we don't need it yet. I'd prefer to hold off until we have running code that uses this. If we were releasing `mapval` as a separate library I'd agree in a heartbeat *rimshot*, but I prefer to have features come from real-world extraction rather than prediction.

It's not a big deal though, since we will probably use it sooner rather than later, so if you feel it's important I'm glad to add it :)